IBM/ibm-common-service-operator
# z specific changes to preload process

Implementing the workaround for cpd quality repo issue 9990. 

Workaround is as follows:

- Scale down mongo operator in ibm-common-services to 0
-  edit icp-mongodb cm in ibm-common-services . Change the net.ssl.mode value to preferSSL from requireSSL
-  restart mongo pods in ibm-common-services one by one starting with icp-mongo-0. Delete the pod, let it come all the way ready, then move on to the next pod
-  clone the github repo for scripts branch, edit the preload_data.sh to remove the ssl parameters from the backup job definition [here](https://github.com/IBM/ibm-common-service-operator/blob/scripts/preload_data.sh#L317) and the restore job definition [here](https://github.com/IBM/ibm-common-service-operator/blob/scripts/preload_data.sh#L490)
-  remove the metrics container from the statefulset yaml (remove from https://github.com/IBM/ibm-common-service-operator/blob/scripts/preload_data.sh#L1554 through https://github.com/IBM/ibm-common-service-operator/blob/scripts/preload_data.sh#L1674)
-  save edited file
-  delete the dummy mongo statefulset in the services namespace (zen in the cluster from this issue)
-  rerun the edited preload_data.sh script with the same parameters
-  after completion, revert change to icp-mongodb configmap in ibm-common-services (aka the original cs namespace) to set net.ssl.mode to requireSSL
-  delete mongo pods one by one again same as before
-  scale mongo operator back to 1

Testing:
- setup cs on z cluster
- run preload_data.sh
- verify script completes and data is transferred into new namespace
